### PR TITLE
Add .travis.yml to run lint and resolve existing lint issues.

### PR DESCRIPTION
**Pull request by Sam1301**
This pr addresses #271 to run lint on debug build.

- Added .travis.yml to run `./gradlew lintDebug` and `./gradlew assembleDebug` 
- Resolved existing lint issues (errors and warnings) in debug build.

I have ignored hardcoded-strings lint error for now, once this #260 is merged, we can change the severity to error again.

### CODE REVIEW

**Comment by smarx** [comment]
*Automated message from Dropbox CLA bot*

@Sam1301, it looks like you've already signed the Dropbox CLA. Thanks!

**Comment by timabbott** [comment]
Awesome!  I enabled Travis CI on this repo.  @saketkumar95 or @kunall17, can one of you give this a quick review?  Would love to merge it soon.

**Comment by timabbott** [comment]
@saketkumar95 @kunall17 ping on reviewing this

**Comment by saketkumar95** [comment]
@timabbott Yeah,  I'll do this today. :)

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    - extra-google-m2repository
+    - extra-android-m2repository
+
+    # uncomment when we enable espresso tests
+    # - sys-img-armeabi-v7a-android-21
+
+  licenses:
+   - 'android-sdk-license-.+'
+
+# uncomment the following lines to use emulator while runnning expresso tests
```
Comment (kunall17) [inline .travis.yml]: A little typo here, <strike>expresso</strike> espresso

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+# Emulator Management: Create, Start and Wait
+#before_script:
+#  - echo no | android create avd --force -n test -t android-21 --abi armeabi-v7a
+#  - emulator -avd test -no-skin -no-audio -no-window &
+#  - android-wait-for-emulator
+#  - adb shell input keyevent 82 &
```
Comment (kunall17) [inline .travis.yml]: What about the server URL?

**Comment by kunall17** [comment]
Can you split the commit, one for the travis setup and one for fixing lint issues?

File: .travis.yml (hunk 2 above)
Comment (Sam1301) [inline .travis.yml]: I am not sure about the context here. Are you suggesting to add server url to `.travis.yml` ?

File: .travis.yml (hunk 1 above)
Comment (Sam1301) [inline .travis.yml]: changed :+1:

**Comment by Sam1301** [comment]
Thanks @kunall17 , split into 2 commits and resolved merge conflict. I have also changed the severity of unused resources to error.

**Comment by timabbott** [comment]
Merged, thanks @Sam1301!!

**Pull request closed without merging**